workers/unity/Assets/Playground/Scripts/Cubes/ScoreCollision.cs

```diff
@@ -0,0 +1,59 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Entities;
+using UnityEngine;
+
+[RemoveAtEndOfTick]
+public struct CollisionComponent : IComponentData
+{
+    public SpatialOSLaunchable own;
+    public SpatialOSLaunchable other;
+}
+
+public class ScoreCollision : MonoBehaviour
+{
+    private MutableView view;
+    private EntityManager entityManager;
+
+    public void Start()
```

Comment:
Better to run this in OnEnable. Right now Start will work fine, but if you pool the prefabs this will break.

Reply:
ACK

Reply:
Update: can't really do this in OnEnable

Reply:
why?

Reply:
The component may not exist at that stage (and hence it's impossible to get the EntityManager). Observed this in practice.

Reply:
you could initialize  lazily instead of during Start or OnEnable. that would also get rid of reassigning `component` during each collision

Reply:
True, though it means we make the check over and over again. It's not a huge change. though.

Reply:
you already make the check `if (entityManager ==null)` already over and over again. just put the logic in there instead of just returning